sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -391,30 +391,22 @@ impl Plan {
                     (
                         // XXXNTP - these boundary servers need a path to the
                         // external network via OPTE.
-                        vec![
-                            ServiceType::Ntp {
-                                servers: config.ntp_servers.clone(),
-                                boundary: true,
-                            },
-                            ServiceType::DnsClient {
-                                servers: config.dns_servers.clone(),
-                                domain: None,
-                            },
-                        ],
+                        vec![ServiceType::Ntp {
+                            ntp_servers: config.ntp_servers.clone(),
+                            boundary: true,
+                            dns_servers: config.dns_servers.clone(),
+                            domain: None,
+                        }],
                         ServiceName::BoundaryNTP,
                     )
                 } else {
                     (
-                        vec![
-                            ServiceType::Ntp {
-                                servers: boundary_ntp_servers.clone(),
-                                boundary: false,
-                            },
-                            ServiceType::DnsClient {
-                                servers: rack_dns_servers.clone(),
-                                domain: None,
-                            },
-                        ],
```

Comment:
This is a reasonable "summary" of this PR -- whenever we ask for the NTP service, the "DNS client" service is now implied.